doc/y_cable/design_doc.md

```diff
@@ -113,105 +143,119 @@ Vendors can have several implementations/ways to use this concept
 
   - Each module of the Y cable vendor can be a class (of each transceiver type) and all we need to do is instantiate the objects of these classes as class instances and these objects will provide the interface of calling the API's for the appropriate vendor Y cable.
   - This instantiation will be done inside xcvrd, when xcvrd starts
-  - These objects basically emulate Y cable instances and whatever action/interaction needs to be done with the Ycable the methods of these objects would provide that
+  - These objects basically emulate Y cable instances and whatever action/interaction needs to be done with the YCable the methods of these objects would provide that
   - each vendor in their implementation can inherit from a base class where there will be definitions for all the supported capabilities of the Y-cable.
-  - If the vendor does not support or implement the utility, it will just raise a not implemented error from the base class itself
+  - for vendors the recommneded approach in case their subclass implementation does not support a method, it should set the method equal to None. This differentiates it from a method they forgot to implement. Then, the calling code should first check if the method is None before attempting to call it.
 
-For example a typical module of the vendor can be like this
-```
+For example the base class would be like this
 
-	class Ycable(Ycablebase):
-            #all vendor modules inherit from Ycablebase
-	    def __init__(self):
+```python
+	class YCableBase(object):
+	    def __init__(self, port):
+		self.port = port
+                <function body here>
 
 	    def toggle_mux_to_torA(self, port):
-		#implement or raise exception if not implemented
-		raise NotImplementedError
+                <function body here>
+
 	    def toggle_mux_to_torB(self, port):
-		#implement or raise exception if not implemented
-		raise NotImplementedError
+                <function body here>
 
 	    def check_prbs(self, port):
-		#implement or raise exception if not implemented
-		raise NotImplementedError
+                <function body here>
 
 	    def only_credo(self, port):
-		raise NotImplementedError
+                <function body here>
 
 	    def only_amphenol(self, port):
-		raise NotImplementedError
+                <function body here>
 
-```
-the base class would be like this
 
 ```
-	class Ycablebase(object):
-	    def __init__(self, port):
-		self.port = port
+
+For example a typical module of the vendor can be like this
+```python
+
+	class YCable(YCableBase):
+            #All vendor modules inherit from YCableBase
```

Comment:
Remove comment here. This should be explained in the "Proposed Solution" description above, no need to add to this example.

Reply:
fixed